tests/test_router_events.py

```diff
@@ -85,3 +82,28 @@ def test_router_events():
     assert state.app_shutdown is True
     assert state.router_shutdown is True
     assert state.sub_router_shutdown is True
+
+
+def test_app_lifespan_state(state: State) -> None:
+    @asynccontextmanager
+    async def lifespan(app: Starlette) -> AsyncIterator[None]:
```

Comment:
This would be a `AsynGenerator`, no🤔? Because of the `yield` in this function.

Reply:
ref in code: https://github.com/tiangolo/fastapi/blob/f67b19f0f73ebdca01775b8c7e531e51b9cecfae/fastapi/concurrency.py#L19

Reply:
According to [StackOverflow](https://stackoverflow.com/questions/63125259/what-is-the-proper-way-to-type-hint-the-return-value-of-an-asynccontextmanager), `AsyncIterator` is also correct as a generator is an iterator, so it is correct and also less complex to type.

Reply:
Every generator is an iterator, but not every iterator is a generator. The point of typing in python is to provide information on the object type as precise as possible. If your take your argument to the extreme, everything could be typed as `object` and it would be technically correct, but not really useful. 

I don't agree with "less complex to type", it is literally one argument more (as generators have a 'send' type and a 'yield' type)?

Reply:
You are right, I made the change.